File: java/main/java/Calculator.java

public class Calculator {

    private static Map<String, (Integer, Integer) -> Integer> supportedOperators = new HashMap(
        ("add", (a, b) -> a + b),
        ("multiply", (a, b) -> a * b),
        ("divide", (a, b) -> a / b),
        ("substract", (a, b) -> a - b)
    )

    public Try<Integer> calculate(Integer a, Integer b, String operator) {

        switch(this.supportedOperators.get(operator)) {
            
        }


    }



}

File: php/src/Calculator.php
<?php

class Calculator {

  private $supportedOperators;

  public function __construct() {
    $this->supportedOperators = [
      "add"       => function ($a, $b) { return $a + $b; },
      "multiply"  => function ($a, $b) { return $a * $b; },
      "divide"    => function ($a, $b) { return $a / $b; },
      "substract" => function ($a, $b) { return $a - $b; }
    ];
  }

  public function calculate($a, $b, $operator) {
    if(!array_key_exists($operator, $this->supportedOperators)) {
      throw new Exception("Unsupported operator {$operator}");
    }
    return $this->supportedOperators[$operator]($a, $b);
  }

}

?>